### PR TITLE
refactor: Move result instead of copy in Solver()

### PR DESCRIPTION
**Pull request by promag**
Move vector `witnessprogram` instead of copying. Also drop unnecessary `std::vector::clear()`.

### CODE REVIEW

File: src/script/standard.cpp [hunk 1]
```diff
@@ -152,7 +152,6 @@ txnouttype Solver(const CScript& scriptPubKey, std::vector<std::vector<unsigned
         return TX_MULTISIG;
     }
 
-    vSolutionsRet.clear();
```
Comment (MarcoFalke) [inline src/script/standard.cpp]: The `clear` here is for documentation purposes (and clearing an empty vector is free anyway)

**Comment by MarcoFalke** [comment]
How much are the speedups?

**Comment by promag** [comment]
@MarcoFalke didn't measured, but think there's no noticeable speedup. I'm just doing the same as [L120](https://github.com/bitcoin/bitcoin/pull/16019/files#diff-1297c2ae09a3b203cb6fa218fa311b0dR120).

File: src/script/standard.cpp (hunk 1 above)
Comment (promag) [reply src/script/standard.cpp]: If it's documentation then maybe change to `assert(vSolutionsRet.empty())`?

Comment (Empact) [reply src/script/standard.cpp]: Would suggest keeping the line - it enforces an invariant against future change, and would be good to consider it separately if we are to change it. `assert` could be overkill given this is basically a failure case anyway.

Comment (jb55) [reply src/script/standard.cpp]: yeah but assert would catch a situation where you missed a return higher up which is something that would definitely be good to catch right away.

Comment (promag) [reply src/script/standard.cpp]: Now using `assert(...)`. It's an unrelated change and I don't mind dropping it but either way `assert()` looks better to me.

**Comment by laanwj** [comment]
I'd prefer to keep this code simpler if there's no noticable speedup.

**Comment by promag** [comment]
@laanwj not even replacing `clear()` with the `assert()`?

**Pull request closed without merging**